### PR TITLE
Extend1726

### PR DESCRIPTION
**Pull request by mirraaj**
Add feature to Laplace Distribution

### CODE REVIEW

**Comment by mirraaj** [comment]
@zoq @rcurtin @ShikharJ Could you please help me with tests? I am unsure of how to test these functions because there isn't any test available for such function with Regression and Laplace distribution.

**Comment by mirraaj** [comment]
@shikharbhardwaj @zoq @rcurtin Could you please help me with the travis build. I am not sure of the reason for it to be failing. Thanks

**Comment by mirraaj** [comment]
@rcurtin @zoq @ShikharJ For this PR I have removed the implementation of Probability and LogProbability function for Regression distribution. I have been trying for quite a few days but was unable to test the file on Travis server. It confuses me because the same tests passed all the test on my personal computer. I think I will give some more time to determine the error for regression distribution.

Currently, this PR focuses on the addition of such function to Laplace distribution. I have added tests for the same and I hope to get this merged soon.

Please let me know if you require any details.

**Comment by zoq** [comment]
Looks like the travis build passed, or maybe I missed something?

**Comment by mirraaj** [comment]
@zoq I am talking about the commit ```Small Change```. If you check the Travis results, you will find that not all the tests have passed. The following link might be useful to you.

https://travis-ci.org/mlpack/mlpack/builds/500615899?utm_source=github_status&utm_medium=notification

I think there was some bug encountered while adding the feature to the regression distribution. To deal with that separately, I decided to split this work into two PR. This current PR focuses on Laplace distribution. Once this is approved I will open another PR to add feature function to regression distribution.

Thanks

**Comment by mirraaj** [comment]
@zoq @rcurtin @ShikharJ This PR is ready to be merged. Please go ahead and approve the changes. There are some issues with regression distribution.I will open another PR to tackle issues for regression data, once this gets merged. Thanks

File: src/mlpack/tests/distribution_test.cpp [hunk 1]
```diff
@@ -1124,6 +1124,58 @@ BOOST_AUTO_TEST_CASE(LaplaceDistributionTest)
...
+  l.Probability(points, probabilities);
+
+  BOOST_REQUIRE_EQUAL(probabilities.n_elem, 2);
+
+  BOOST_REQUIRE_CLOSE(probabilities(0), 0.500000000000000, 1e-5);
+  BOOST_REQUIRE_CLOSE(probabilities(1), 0.183939720585721, 1e-5);
+}
+
+/**
+ * Laplace Distribution Log Probability Test
```
Comment (ShikharJ) [inline src/mlpack/tests/distribution_test.cpp]: Same here.

File: src/mlpack/core/dists/laplace_distribution.hpp [hunk 2]
```diff
@@ -158,6 +179,17 @@ class LaplaceDistribution
   double scale;
 };
 
+inline void LaplaceDistribution::LogProbability(
+    const arma::mat& x,
+    arma::vec& logProbabilities) const
+{
```
Comment (zoq) [inline src/mlpack/core/dists/laplace_distribution.hpp]: I guess, we could also put this inside the class itself, this is what was done for the ``Random`` method.

Comment (rcurtin) [reply src/mlpack/core/dists/laplace_distribution.hpp]: I agree, that would be an improvement.  I'll do it during merge.

**Comment by mirraaj** [comment]
I will do it and update it soon

**Comment by rcurtin** [comment]
@mirraaj no worries, already took care of it. :+1:  Thanks for the contribution!

**Pull request merged**